server/rest/s3_dataset.py

```diff
@@ -122,3 +127,37 @@ def createDataset(self, item, params):
         and trigger import.')
         .param('id', 'The Item ID', paramType='path')
         .errorResponse('ID was invalid.'))
+
+
+    @access.public
+    @loadmodel(model='item', level=AccessType.WRITE)
+    def updateDataset(self, item, params):
+
+        # de-stringify selectedItems
+        params['selectedItems'] = json_util.loads(params.get("selectedItems", "[]"))
+        params['folderId'] = ObjectId(params['folderId'])
+
+        # Metadata keys we allow to be updated
+        valid_keys = ['selectedItems']
+
+        try:
+            minerva_metadata = item['meta']['minerva']
+        except KeyError:
+            raise RestException(u"No metadata set, is this really a dataset?")
+
+        # Strictly allow only updates to keys in valid_keys
+        if len(set([i for i in minerva_metadata.items() if i[0] not in valid_keys]) ^
+               set([i for i in params.items() if i[0] not in valid_keys])) > 0:
+            raise RestException("Only %s may be updated!" % valid_keys)
+
```

Comment:
Probably could be simplified (or better named?),  the idea was to white list values that can be updated through this method and disallow all other metadata values to changed.

keys like assetstoreId and and original_type shouldn't be allowed to change correct?